### PR TITLE
Update Continuous Integration HOL for new VSTS version with hosted agent

### PR DESCRIPTION
**Pull request by sawpresto**
Updated lab so that it can use a hosted VSTS agent, updated images so that they reflected the new VSTS version.

### CODE REVIEW

File: docs/HOL_Continuous-Integration/README.md [hunk 1]
```diff
@@ -1,215 +1,38 @@
...
 master branch, we want to ensure that it integrates into our code correctly to
-get fast feedback. To do so, we are going to be setting up a build agent that
+get fast feedback. To do so, we are going to be creating a build definition that
 will allow us to compile and run unit tests on our code every time a commit is
 pushed to Visual Studio Team Services.
 
 ###Pre-requisites:###
 
--   An active Visual Studio Team Services (VSTS) account
+-   Completion of the "Set up Parts Unlimited MRP" lab
```
Comment (julienstroheker) [inline docs/HOL_Continuous-Integration/README.md]: Add the hyperlink to the HOL when the merge will done

File: docs/HOL_Continuous-Integration/README.md [hunk 2]
```diff
@@ -1,215 +1,38 @@
...
-	sudo add-apt-repository ppa:openjdk-r/ppa
-
-	sudo apt-get update
-
-**8.** Copy and paste the following snippet to **run these commands**:
+>NOTE: Ensure that you have an existing PartsUnlimitedMRP team project that also contains the Git repository cloned from GitHub. If not, complete the "Set up Parts Unlimited MRP" lab before going through this lab. 
 
-	# Install git client
-	sudo apt-get install git -y
+>In this lab, we will be using the Hosted agent located in Visual Studio Team Services. If you would like to use an on-premises cross-platform agent (Azure subscription needed), you can follow instructions for setting an agent up [with this link](https://github.com/Microsoft/vsts-agent/blob/master/README.md).
```
Comment (julienstroheker) [inline docs/HOL_Continuous-Integration/README.md]: The link redirect to the repo for VSTS Agent not to the instructions for setting an agent up....

File: docs/HOL_Continuous-Integration/README.md [hunk 3]
```diff
@@ -219,116 +42,116 @@ the page.
...
-*OrderService* and set the **Gradle Wrapper** to the following location:
-(NOTE: set the Options to **-x text** as the test have external dependencies on a mongo database.)
+**8.** Select the second Gradle task and **click the pencil icon** to edit the task name. Name the task *OrderService* and set the **Gradle Wrapper** to the following location:
 
 	src/Backend/OrderService/gradlew
-	-x test
+
+![](<media/build_gradle_order.png>)
 	
+**9.** Since Order Service does have unit tests in the project, we can automate running the tests as part of the build by adding in `test` in the **Gradle tasks** field. Keep the **JUnit Test Results** checkbox to "Publish to VSO/TFS" checked, and set the **Test Results Files** field to `**/TEST-*.xml`.
```
Comment (julienstroheker) [inline docs/HOL_Continuous-Integration/README.md]: Typo : 
```checkbox to "Publish to VSO/TFS" checked```
change for 
```checkbox to "Publish to TFS/Team Services" checked```

File: docs/HOL_Continuous-Integration/README.md [hunk 4]
```diff
@@ -219,116 +42,116 @@ the page.
...
-and set the **Gradle Wrapper** to the following location:
+**10.** Select the third Gradle task and **click the pencil icon** to edit the task name. Name the task *Clients* and set the **Gradle Wrapper** to the following location:
 
 	src/Clients/gradlew
 
-Set the **Working Directory** to the following location:
+![](<media/build_gradle_clients.png>)
 
-	src/Clients
+**11.** Uncheck the checkbox in **JUnit Test Results** to Publish to VSO/TFS since we will not be running automated tests in Clients. Expand the **Advanced** section, and set the **Working Directory** to the following location:
```
Comment (julienstroheker) [inline docs/HOL_Continuous-Integration/README.md]: Same Typo : to Publish to VSO/TFS

File: docs/HOL_Continuous-Integration/README.md [hunk 5]
```diff
@@ -219,116 +42,116 @@ the page.
...
-*IntegrationService* and set the **Gradle Wrapper** to the following location:
+**6.** Select the first Gradle task and **click the pencil icon** to edit the task name. Name the task *IntegrationService* and set the **Gradle Wrapper** to the following location (either type or browse using the **...** button):
 
-	src/Backend/IntegrationService/gradlew 
+	src/Backend/IntegrationService/gradlew
 
-Set the **Working Directory** to the following location:
+![](<media/build_gradle_integration.png>)
+
+**7.** Uncheck the checkbox in **JUnit Test Results** to Publish to VSO/TFS since we will not be running automated tests in the Integration Service. Expand the **Advanced** section, and set the **Working Directory** to the following location:
```
Comment (julienstroheker) [inline docs/HOL_Continuous-Integration/README.md]: Typo : 
```to Publish to VSO/TFS ```
change for 
```to Publish to TFS/Team Services```

File: docs/HOL_Continuous-Integration/README.md [hunk 6]
```diff
@@ -219,116 +42,116 @@ the page.
...
-	Contents: **/build/libs/!(buildSrc)*.?ar
+	Copy Root: $(Build.SourcesDirectory)
+    Contents: **/build/libs/!(buildSrc)*.?ar
 	Artifact Name: drop
 	Artifact Type: Server
 
 ![](<media/build_pub_step_details.png>)
 
-**NOTE:** The Content field supports minimatch patterns. You can find more information here: [https://github.com/Microsoft/vso-agent-tasks](https://github.com/Microsoft/vso-agent-tasks "https://github.com/Microsoft/vso-agent-tasks")
+**11.** Select the second Publish Build Artifacts task, and fill in the input values with the following:
```
Comment (julienstroheker) [inline docs/HOL_Continuous-Integration/README.md]: Do we need this part in this HOL ?
My guess is to talk about that in the CD HOL

File: docs/HOL_Continuous-Integration/README.md [hunk 7]
```diff
@@ -219,116 +42,116 @@ the page.
...
+
+![](<media/second_copy_publish.png>)
 
-**12.** Go to the **Triggers** tab and **select Continuous Integration (CI)**
+**12.** Go to the **Triggers** tab and **select Continuous Integration (CI)**.
 
 ![](<media/build_ci_trigger.png>)
 
-**13.** Click **General**, set the default queue to the previously created queue (**linux**)
+**13.** Click **General**, set the default queue to the apprpriate queue. Leave as the **Hosted** queue from the previous steps.
```
Comment (julienstroheker) [inline docs/HOL_Continuous-Integration/README.md]: apprpriate --> appropriate ?

**Comment by julienstroheker** [comment]
When I am using a Hosted Agent, I have all the test related to Mongo failing and it took 5 minute to run the build : 
```
:testClasses
:test
smpl.ordering.repositories.mongodb.test.MongoCatalogItemsRepositoryTest > testRemoveCatalogItem FAILED
com.mongodb.MongoTimeoutException at MongoCatalogItemsRepositoryTest.java:22
smpl.ordering.repositories.mongodb.test.MongoCatalogItemsRepositoryTest > testGetCatalogItems FAILED
com.mongodb.MongoTimeoutException at MongoCatalogItemsRepositoryTest.java:22
smpl.ordering.repositories.mongodb.test.MongoCatalogItemsRepositoryTest > testGetCatalogItem FAILED
com.mongodb.MongoTimeoutException at MongoCatalogItemsRepositoryTest.java:22
smpl.ordering.repositories.mongodb.test.MongoCatalogItemsRepositoryTest > testUpsertCatalogItem FAILED
com.mongodb.MongoTimeoutException at MongoCatalogItemsRepositoryTest.java:22
smpl.ordering.repositories.mongodb.test.MongoDealersRepositoryTest > testUpsertDealer FAILED
org.springframework.dao.DataAccessResourceFailureException at MongoDealersRepositoryTest.java:24
Caused by: com.mongodb.MongoTimeoutException at MongoDealersRepositoryTest.java:24
smpl.ordering.repositories.mongodb.test.MongoDealersRepositoryTest > testRemoveDealer FAILED
org.springframework.dao.DataAccessResourceFailureException at MongoDealersRepositoryTest.java:24
Caused by: com.mongodb.MongoTimeoutException at MongoDealersRepositoryTest.java:24
smpl.ordering.repositories.mongodb.test.MongoDealersRepositoryTest > testGetDealer FAILED
org.springframework.dao.DataAccessResourceFailureException at MongoDealersRepositoryTest.java:24
Caused by: com.mongodb.MongoTimeoutException at MongoDealersRepositoryTest.java:24
smpl.ordering.repositories.mongodb.test.MongoDealersRepositoryTest > testGetDealers FAILED
org.springframework.dao.DataAccessResourceFailureException at MongoDealersRepositoryTest.java:24
Caused by: com.mongodb.MongoTimeoutException at MongoDealersRepositoryTest.java:24
smpl.ordering.repositories.mongodb.test.MongoOrderRepositoryTest > testGetOrder FAILED
org.springframework.dao.DataAccessResourceFailureException at MongoOrderRepositoryTest.java:23
Caused by: com.mongodb.MongoTimeoutException at MongoOrderRepositoryTest.java:23
smpl.ordering.repositories.mongodb.test.MongoOrderRepositoryTest > testGetOrdersByDealerName FAILED
org.springframework.dao.DataAccessResourceFailureException at MongoOrderRepositoryTest.java:23
Caused by: com.mongodb.MongoTimeoutException at MongoOrderRepositoryTest.java:23
smpl.ordering.repositories.mongodb.test.MongoOrderRepositoryTest > testUpdateOrder FAILED
org.springframework.dao.DataAccessResourceFailureException at MongoOrderRepositoryTest.java:23
Caused by: com.mongodb.MongoTimeoutException at MongoOrderRepositoryTest.java:23
smpl.ordering.repositories.mongodb.test.MongoOrderRepositoryTest > testCreateOrder FAILED
org.springframework.dao.DataAccessResourceFailureException at MongoOrderRepositoryTest.java:23
Caused by: com.mongodb.MongoTimeoutException at MongoOrderRepositoryTest.java:23
smpl.ordering.repositories.mongodb.test.MongoOrderRepositoryTest > testUpdateOrder1 FAILED
org.springframework.dao.DataAccessResourceFailureException at MongoOrderRepositoryTest.java:23
Caused by: com.mongodb.MongoTimeoutException at MongoOrderRepositoryTest.java:23
smpl.ordering.repositories.mongodb.test.MongoOrderRepositoryTest > testGetOrdersByStatus FAILED
org.springframework.dao.DataAccessResourceFailureException at MongoOrderRepositoryTest.java:23
Caused by: com.mongodb.MongoTimeoutException at MongoOrderRepositoryTest.java:23
smpl.ordering.repositories.mongodb.test.MongoOrderRepositoryTest > testGetOrdersByQuoteId FAILED
org.springframework.dao.DataAccessResourceFailureException at MongoOrderRepositoryTest.java:23
Caused by: com.mongodb.MongoTimeoutException at MongoOrderRepositoryTest.java:23
smpl.ordering.repositories.mongodb.test.MongoOrderRepositoryTest > testHasOrder FAILED
org.springframework.dao.DataAccessResourceFailureException at MongoOrderRepositoryTest.java:23
Caused by: com.mongodb.MongoTimeoutException at MongoOrderRepositoryTest.java:23
smpl.ordering.repositories.mongodb.test.MongoQuoteRepositoryTest > testGetQuote FAILED
org.springframework.dao.DataAccessResourceFailureException at MongoQuoteRepositoryTest.java:23
Caused by: com.mongodb.MongoTimeoutException at MongoQuoteRepositoryTest.java:23
smpl.ordering.repositories.mongodb.test.MongoQuoteRepositoryTest > testUpdateQuote FAILED
org.springframework.dao.DataAccessResourceFailureException at MongoQuoteRepositoryTest.java:23
Caused by: com.mongodb.MongoTimeoutException at MongoQuoteRepositoryTest.java:23
smpl.ordering.repositories.mongodb.test.MongoQuoteRepositoryTest > testCreateQuote FAILED
org.springframework.dao.DataAccessResourceFailureException at MongoQuoteRepositoryTest.java:23
Caused by: com.mongodb.MongoTimeoutException at MongoQuoteRepositoryTest.java:23
smpl.ordering.repositories.mongodb.test.MongoQuoteRepositoryTest > testRemoveQuote FAILED
org.springframework.dao.DataAccessResourceFailureException at MongoQuoteRepositoryTest.java:23
Caused by: com.mongodb.MongoTimeoutException at MongoQuoteRepositoryTest.java:23
smpl.ordering.repositories.mongodb.test.MongoQuoteRepositoryTest > testGetQuotesByCustomerName FAILED
org.springframework.dao.DataAccessResourceFailureException at MongoQuoteRepositoryTest.java:23
Caused by: com.mongodb.MongoTimeoutException at MongoQuoteRepositoryTest.java:23
smpl.ordering.repositories.mongodb.test.MongoShipmentRepositoryTest > testGetShipments FAILED
org.springframework.dao.DataAccessResourceFailureException at MongoShipmentRepositoryTest.java:23
Caused by: com.mongodb.MongoTimeoutException at MongoShipmentRepositoryTest.java:23
smpl.ordering.repositories.mongodb.test.MongoShipmentRepositoryTest > testUpdateShipment FAILED
org.springframework.dao.DataAccessResourceFailureException at MongoShipmentRepositoryTest.java:23
Caused by: com.mongodb.MongoTimeoutException at MongoShipmentRepositoryTest.java:23
smpl.ordering.repositories.mongodb.test.MongoShipmentRepositoryTest > testCreateShipment FAILED
org.springframework.dao.DataAccessResourceFailureException at MongoShipmentRepositoryTest.java:23
Caused by: com.mongodb.MongoTimeoutException at MongoShipmentRepositoryTest.java:23
smpl.ordering.repositories.mongodb.test.MongoShipmentRepositoryTest > testAddEvent FAILED
org.springframework.dao.DataAccessResourceFailureException at MongoShipmentRepositoryTest.java:23
Caused by: com.mongodb.MongoTimeoutException at MongoShipmentRepositoryTest.java:23
smpl.ordering.repositories.mongodb.test.MongoShipmentRepositoryTest > testGetShipmentById FAILED
org.springframework.dao.DataAccessResourceFailureException at MongoShipmentRepositoryTest.java:23
Caused by: com.mongodb.MongoTimeoutException at MongoShipmentRepositoryTest.java:23
21:24:14.248 [Thread-38] DEBUG o.a.h.i.c.PoolingHttpClientConnectionManager - Connection manager is shutting down
21:24:14.254 [Thread-38] DEBUG o.a.h.i.c.PoolingHttpClientConnectionManager - Connection manager shut down
79 tests completed, 26 failed
:test FAILED
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///C:/a/1/s/src/Backend/OrderService/build/reports/tests/index.html
* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
BUILD FAILED
Total time: 5 mins 8.949 secs
System.Exception: Unexpected exit code received from batch file: 1
at Microsoft.TeamFoundation.DistributedTask.Task.Internal.PowerShell.InvokeBatchScriptCmdlet.ProcessRecord()
at System.Management.Automation.CommandProcessor.ProcessRecord()
Publishing Test Results...
Test results remaining: 79
Published Test Run : https://julienstroheker.visualstudio.com/PartsUnlimitedReview/_TestManagement/Runs#runId=101&_a=runCharts
******************************************************************************
Finishing task: Gradle
******************************************************************************
System.Exception: Task Gradle failed. This caused the job to fail. Look at the logs for the task for more details.
at Microsoft.TeamFoundation.DistributedTask.Worker.JobRunner.Run(IJobContext jobContext, IJobRequest job, IJobExtension jobExtension, CancellationTokenSource tokenSource)
******************************************************************************
Finishing Build
******************************************************************************
Worker Worker-2b338f96-5661-468b-8c69-692cd5b81d8d finished running job 2b338f96-5661-468b-8c69-692cd5b81d8d
```

**Comment by sawpresto** [comment]
Right, yep, we got that early on as well because Gradle is trying to run integration tests as unit tests. We made a fix to that and I must not have committed that. I'll update a bit of the code and push the changes tonight!

**Comment by julienstroheker** [comment]
@sawpresto Thanks you very much !

**Comment by sawpresto** [comment]
Pushed the commit that should fix the MongoDB tests from running. Let me know if it still gives you issues.

**Comment by julienstroheker** [comment]
I have an other error : 
```
******************************************************************************
Starting task: gradlew build test
******************************************************************************
Executing the powershell script: C:\LR\MMS\Services\Mms\TaskAgentProvisioner\Tools\agents\1.103.2\tasks\Gradle\1.0.48\gradle.ps1
SonarQube analysis is not yet supported on Windows and hosted agents.
Please use a dedicated Unix agent to enable this feature: https://github.com/Microsoft/vsts-agent
PMD code analysis is not yet supported on Windows and hosted agents.
Please use a dedicated Unix agent to enable this feature: https://github.com/Microsoft/vsts-agent
:buildSrc:compileJava UP-TO-DATE
:buildSrc:compileGroovy UP-TO-DATE
:buildSrc:processResources UP-TO-DATE
:buildSrc:classes UP-TO-DATE
:buildSrc:jar UP-TO-DATE
:buildSrc:assemble UP-TO-DATE
:buildSrc:compileTestJava UP-TO-DATE
:buildSrc:compileTestGroovy UP-TO-DATE
:buildSrc:processTestResources UP-TO-DATE
:buildSrc:testClasses UP-TO-DATE
:buildSrc:test UP-TO-DATE
:buildSrc:check UP-TO-DATE
:buildSrc:build UP-TO-DATE
:compileJava
Download https://repo1.maven.org/maven2/com/microsoft/azure/applicationinsights-core/0.9.0/applicationinsights-core-0.9.0.pom
Download https://repo1.maven.org/maven2/org/springframework/boot/spring-boot-starter-web/1.1.7.RELEASE/spring-boot-starter-web-1.1.7.RELEASE.pom
Download https://repo1.maven.org/maven2/org/springframework/boot/spring-boot-starters/1.1.7.RELEASE/spring-boot-starters-1.1.7.RELEASE.pom
Download https://repo1.maven.org/maven2/org/springframework/boot/spring-boot-parent/1.1.7.RELEASE/spring-boot-parent-1.1.7.RELEASE.pom
Download https://repo1.maven.org/maven2/org/springframework/boot/spring-boot-starter-actuator/1.1.7.RELEASE/spring-boot-starter-actuator-1.1.7.RELEASE.pom
Download https://repo1.maven.org/maven2/org/springframework/data/spring-data-mongodb/1.5.4.RELEASE/spring-data-mongodb-1.5.4.RELEASE.pom
Download https://repo1.maven.org/maven2/org/springframework/data/spring-data-mongodb-parent/1.5.4.RELEASE/spring-data-mongodb-parent-1.5.4.RELEASE.pom
Download https://repo1.maven.org/maven2/org/springframework/data/build/spring-data-parent/1.4.4.RELEASE/spring-data-parent-1.4.4.RELEASE.pom
Download https://repo1.maven.org/maven2/org/springframework/spring-framework-bom/3.2.10.RELEASE/spring-framework-bom-3.2.10.RELEASE.pom
Download https://repo1.maven.org/maven2/org/apache/httpcomponents/httpcomponents-client/4.3.5/httpcomponents-client-4.3.5.pom
Download https://repo1.maven.org/maven2/org/apache/httpcomponents/project/7/project-7.pom
Download https://repo1.maven.org/maven2/com/microsoft/azure/applicationinsights-core/1.0.5/applicationinsights-core-1.0.5.pom
Download https://repo1.maven.org/maven2/org/springframework/boot/spring-boot-starter/1.1.7.RELEASE/spring-boot-starter-1.1.7.RELEASE.pom
Download https://repo1.maven.org/maven2/org/springframework/boot/spring-boot-starter-tomcat/1.1.7.RELEASE/spring-boot-starter-tomcat-1.1.7.RELEASE.pom
Download https://repo1.maven.org/maven2/com/fasterxml/jackson/core/jackson-databind/2.3.4/jackson-databind-2.3.4.pom
Download https://repo1.maven.org/maven2/com/fasterxml/oss-parent/12/oss-parent-12.pom
Download https://repo1.maven.org/maven2/org/hibernate/hibernate-validator/5.0.3.Final/hibernate-validator-5.0.3.Final.pom
Download https://repo1.maven.org/maven2/org/hibernate/hibernate-validator-parent/5.0.3.Final/hibernate-validator-parent-5.0.3.Final.pom
Download https://repo1.maven.org/maven2/org/jboss/arquillian/arquillian-bom/1.0.2.Final/arquillian-bom-1.0.2.Final.pom
Download https://repo1.maven.org/maven2/org/jboss/shrinkwrap/shrinkwrap-bom/1.0.1/shrinkwrap-bom-1.0.1.pom
Download https://repo1.maven.org/maven2/org/jboss/shrinkwrap/resolver/shrinkwrap-resolver-bom/1.0.0-beta-7/shrinkwrap-resolver-bom-1.0.0-beta-7.pom
Download https://repo1.maven.org/maven2/org/jboss/shrinkwrap/descriptors/shrinkwrap-descriptors-bom/2.0.0-alpha-3/shrinkwrap-descriptors-bom-2.0.0-alpha-3.pom
Download https://repo1.maven.org/maven2/org/springframework/spring-web/4.0.7.RELEASE/spring-web-4.0.7.RELEASE.pom
Download https://repo1.maven.org/maven2/org/springframework/spring-webmvc/4.0.7.RELEASE/spring-webmvc-4.0.7.RELEASE.pom
Download https://repo1.maven.org/maven2/org/springframework/boot/spring-boot-actuator/1.1.7.RELEASE/spring-boot-actuator-1.1.7.RELEASE.pom
Download https://repo1.maven.org/maven2/org/springframework/spring-tx/3.2.10.RELEASE/spring-tx-3.2.10.RELEASE.pom
Download https://repo1.maven.org/maven2/org/springframework/spring-context/3.2.10.RELEASE/spring-context-3.2.10.RELEASE.pom
Download https://repo1.maven.org/maven2/org/springframework/spring-beans/3.2.10.RELEASE/spring-beans-3.2.10.RELEASE.pom
Download https://repo1.maven.org/maven2/org/springframework/spring-expression/3.2.10.RELEASE/spring-expression-3.2.10.RELEASE.pom
Download https://repo1.maven.org/maven2/org/springframework/data/spring-data-commons/1.8.4.RELEASE/spring-data-commons-1.8.4.RELEASE.pom
Download https://repo1.maven.org/maven2/org/slf4j/slf4j-parent/1.7.7/slf4j-parent-1.7.7.pom
Download https://repo1.maven.org/maven2/org/apache/apache/9/apache-9.pom
Download https://repo1.maven.org/maven2/org/springframework/boot/spring-boot/1.1.7.RELEASE/spring-boot-1.1.7.RELEASE.pom
Download https://repo1.maven.org/maven2/org/springframework/boot/spring-boot-autoconfigure/1.1.7.RELEASE/spring-boot-autoconfigure-1.1.7.RELEASE.pom
Download https://repo1.maven.org/maven2/org/springframework/boot/spring-boot-starter-logging/1.1.7.RELEASE/spring-boot-starter-logging-1.1.7.RELEASE.pom
Download https://repo1.maven.org/maven2/org/yaml/snakeyaml/1.13/snakeyaml-1.13.pom
Download https://repo1.maven.org/maven2/org/apache/tomcat/embed/tomcat-embed-core/7.0.55/tomcat-embed-core-7.0.55.pom
Download https://repo1.maven.org/maven2/org/apache/tomcat/embed/tomcat-embed-el/7.0.55/tomcat-embed-el-7.0.55.pom
Download https://repo1.maven.org/maven2/org/apache/tomcat/embed/tomcat-embed-logging-juli/7.0.55/tomcat-embed-logging-juli-7.0.55.pom
Download https://repo1.maven.org/maven2/com/fasterxml/jackson/core/jackson-annotations/2.3.0/jackson-annotations-2.3.0.pom
Download https://repo1.maven.org/maven2/com/fasterxml/oss-parent/11/oss-parent-11.pom
Download https://repo1.maven.org/maven2/com/fasterxml/jackson/core/jackson-core/2.3.4/jackson-core-2.3.4.pom
Download https://repo1.maven.org/maven2/javax/validation/validation-api/1.1.0.Final/validation-api-1.1.0.Final.pom
Download https://repo1.maven.org/maven2/com/fasterxml/classmate/1.0.0/classmate-1.0.0.pom
Download https://repo1.maven.org/maven2/org/slf4j/jul-to-slf4j/1.7.7/jul-to-slf4j-1.7.7.pom
Download https://repo1.maven.org/maven2/org/slf4j/log4j-over-slf4j/1.7.7/log4j-over-slf4j-1.7.7.pom
Download https://repo1.maven.org/maven2/com/google/guava/guava/12.0.1/guava-12.0.1.pom
Download https://repo1.maven.org/maven2/com/google/guava/guava-parent/12.0.1/guava-parent-12.0.1.pom
Download https://repo1.maven.org/maven2/eu/infomas/annotation-detector/3.0.4/annotation-detector-3.0.4.pom
Download https://repo1.maven.org/maven2/eu/infomas/infomas-asl/3.0.4/infomas-asl-3.0.4.pom
Download https://repo1.maven.org/maven2/org/sonatype/oss/oss-parent/9/oss-parent-9.pom
Download https://repo1.maven.org/maven2/commons-io/commons-io/2.4/commons-io-2.4.pom
Download https://repo1.maven.org/maven2/org/apache/commons/commons-parent/25/commons-parent-25.pom
Download https://repo1.maven.org/maven2/org/apache/commons/commons-lang3/3.1/commons-lang3-3.1.pom
Download https://repo1.maven.org/maven2/com/google/code/findbugs/jsr305/1.3.9/jsr305-1.3.9.pom
Download https://repo1.maven.org/maven2/org/jboss/logging/jboss-logging/3.1.1.GA/jboss-logging-3.1.1.GA.pom
Download https://repo1.maven.org/maven2/org/springframework/boot/spring-boot-starter-web/1.1.7.RELEASE/spring-boot-starter-web-1.1.7.RELEASE.jar
Download https://repo1.maven.org/maven2/org/springframework/boot/spring-boot-starter-actuator/1.1.7.RELEASE/spring-boot-starter-actuator-1.1.7.RELEASE.jar
Download https://repo1.maven.org/maven2/org/springframework/data/spring-data-mongodb/1.5.4.RELEASE/spring-data-mongodb-1.5.4.RELEASE.jar
Download https://repo1.maven.org/maven2/org/springframework/boot/spring-boot-starter/1.1.7.RELEASE/spring-boot-starter-1.1.7.RELEASE.jar
Download https://repo1.maven.org/maven2/org/springframework/boot/spring-boot-starter-tomcat/1.1.7.RELEASE/spring-boot-starter-tomcat-1.1.7.RELEASE.jar
Download https://repo1.maven.org/maven2/com/fasterxml/jackson/core/jackson-databind/2.3.4/jackson-databind-2.3.4.jar
Download https://repo1.maven.org/maven2/org/hibernate/hibernate-validator/5.0.3.Final/hibernate-validator-5.0.3.Final.jar
Download https://repo1.maven.org/maven2/org/springframework/spring-web/4.0.7.RELEASE/spring-web-4.0.7.RELEASE.jar
Download https://repo1.maven.org/maven2/org/springframework/spring-webmvc/4.0.7.RELEASE/spring-webmvc-4.0.7.RELEASE.jar
Download https://repo1.maven.org/maven2/org/springframework/boot/spring-boot-actuator/1.1.7.RELEASE/spring-boot-actuator-1.1.7.RELEASE.jar
Download https://repo1.maven.org/maven2/org/springframework/spring-tx/3.2.10.RELEASE/spring-tx-3.2.10.RELEASE.jar
Download https://repo1.maven.org/maven2/org/springframework/data/spring-data-commons/1.8.4.RELEASE/spring-data-commons-1.8.4.RELEASE.jar
Download https://repo1.maven.org/maven2/org/springframework/boot/spring-boot/1.1.7.RELEASE/spring-boot-1.1.7.RELEASE.jar
Download https://repo1.maven.org/maven2/org/springframework/boot/spring-boot-autoconfigure/1.1.7.RELEASE/spring-boot-autoconfigure-1.1.7.RELEASE.jar
Download https://repo1.maven.org/maven2/org/springframework/boot/spring-boot-starter-logging/1.1.7.RELEASE/spring-boot-starter-logging-1.1.7.RELEASE.jar
Download https://repo1.maven.org/maven2/org/yaml/snakeyaml/1.13/snakeyaml-1.13.jar
Download https://repo1.maven.org/maven2/org/apache/tomcat/embed/tomcat-embed-core/7.0.55/tomcat-embed-core-7.0.55.jar
Download https://repo1.maven.org/maven2/org/apache/tomcat/embed/tomcat-embed-el/7.0.55/tomcat-embed-el-7.0.55.jar
Download https://repo1.maven.org/maven2/org/apache/tomcat/embed/tomcat-embed-logging-juli/7.0.55/tomcat-embed-logging-juli-7.0.55.jar
Download https://repo1.maven.org/maven2/com/fasterxml/jackson/core/jackson-annotations/2.3.0/jackson-annotations-2.3.0.jar
Download https://repo1.maven.org/maven2/com/fasterxml/jackson/core/jackson-core/2.3.4/jackson-core-2.3.4.jar
Download https://repo1.maven.org/maven2/javax/validation/validation-api/1.1.0.Final/validation-api-1.1.0.Final.jar
Download https://repo1.maven.org/maven2/com/fasterxml/classmate/1.0.0/classmate-1.0.0.jar
Download https://repo1.maven.org/maven2/org/slf4j/jul-to-slf4j/1.7.7/jul-to-slf4j-1.7.7.jar
Download https://repo1.maven.org/maven2/org/slf4j/log4j-over-slf4j/1.7.7/log4j-over-slf4j-1.7.7.jar
Download https://repo1.maven.org/maven2/com/microsoft/azure/applicationinsights-core/1.0.5/applicationinsights-core-1.0.5.jar
Download https://repo1.maven.org/maven2/com/google/guava/guava/12.0.1/guava-12.0.1.jar
Download https://repo1.maven.org/maven2/eu/infomas/annotation-detector/3.0.4/annotation-detector-3.0.4.jar
Download https://repo1.maven.org/maven2/commons-io/commons-io/2.4/commons-io-2.4.jar
Download https://repo1.maven.org/maven2/org/apache/commons/commons-lang3/3.1/commons-lang3-3.1.jar
Download https://repo1.maven.org/maven2/com/google/code/findbugs/jsr305/1.3.9/jsr305-1.3.9.jar
Download https://repo1.maven.org/maven2/org/jboss/logging/jboss-logging/3.1.1.GA/jboss-logging-3.1.1.GA.jar
C:\a\1\s\src\Backend\OrderService\src\main\java\smpl\ordering\AppInsightsFilter.java:12: error: cannot find symbol
import com.microsoft.applicationinsights.telemetry.HttpRequestTelemetry;
^
symbol: class HttpRequestTelemetry
location: package com.microsoft.applicationinsights.telemetry
C:\a\1\s\src\Backend\OrderService\src\main\java\smpl\ordering\AppInsightsFilter.java:42: error: cannot find symbol
HttpRequestTelemetry telemetry = new HttpRequestTelemetry(String.format("%s %s", method, rURI), startTime, 0L, "200", false);
^
symbol: class HttpRequestTelemetry
location: class AppInsightsFilter
C:\a\1\s\src\Backend\OrderService\src\main\java\smpl\ordering\AppInsightsFilter.java:42: error: cannot find symbol
HttpRequestTelemetry telemetry = new HttpRequestTelemetry(String.format("%s %s", method, rURI), startTime, 0L, "200", false);
^
symbol: class HttpRequestTelemetry
location: class AppInsightsFilter
C:\a\1\s\src\Backend\OrderService\src\main\java\smpl\ordering\repositories\mongodb\MongoOperationsWithRetry.java:180: error: cannot find symbol
rdt.setValue((double) (end.getTime() - start.getTime()));
^
symbol: method setValue(double)
location: variable rdt of type RemoteDependencyTelemetry
C:\a\1\s\src\Backend\OrderService\src\main\java\smpl\ordering\repositories\mongodb\MongoOperationsWithRetry.java:182: error: cannot find symbol
rdt.setDependencyKind(DependencyKind.Undefined);
^
symbol: variable Undefined
location: class DependencyKind
Note: C:\a\1\s\src\Backend\OrderService\src\main\java\smpl\ordering\repositories\mongodb\MongoOperationsWithRetry.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
5 errors
:compileJava FAILED
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':compileJava'.
> Compilation failed; see the compiler error output for details.
* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
BUILD FAILED
Total time: 15.47 secs
System.Exception: Unexpected exit code received from batch file: 1
at Microsoft.TeamFoundation.DistributedTask.Task.Internal.PowerShell.InvokeBatchScriptCmdlet.ProcessRecord()
at System.Management.Automation.CommandProcessor.ProcessRecord()
No JUnit test results files were found matching pattern 'C:\a\1\s\**\TEST-*.xml', so publishing JUnit test results is being skipped.
******************************************************************************
Finishing task: gradlew 
******************************************************************************
```

**Comment by sawpresto** [comment]
Just added a new commit to fix the AI custom telemetry code, but it's also in the APM HOL, I believe.

**Comment by julienstroheker** [comment]
@sawpresto It works on my side ! Thanks a lot ! 
Can you check the others issues inside the other PR ?
Thanks a lot !

**Pull request merged**